internal/temperature/temperature.go

```diff
@@ -0,0 +1,116 @@
+package temperature
+
+import (
+	"bufio"
+	"fmt"
+	"gobar/internal/blockutils"
+	"gobar/internal/clickutils"
+	"gobar/internal/log"
+	"io/ioutil"
+	"os"
+	"regexp"
+	"strconv"
+	"time"
+)
+
+func getTempFromPath(path string) float64 {
+	var tmpBytes []byte
+	val := 0.0
+	f, err := os.Open(path)
+	if err != nil {
+		log.FileLog(err)
+	}
+	defer f.Close()
+
+	sc := bufio.NewScanner(f)
+	sc.Scan()
+
+	tmpBytes = sc.Bytes()
+
+	val, err = strconv.ParseFloat(string(tmpBytes), 32)
+	if err != nil {
+		log.FileLog(err)
+		val = 0.0
+	}
+
+	return val / 1000.0
+}
+
+func findHWMon() string {
+	platformDir := `/sys/devices/platform/`
+	coretempRegex := regexp.MustCompile(`coretemp\.[0-9]+`)
```

Comment:
i like to put regex compilation into global vars so this code isn't run repeatedly